File: hibernate-scala-examples/src/main/scala/org/hibernate/examples/hibernate/interceptor/UpdatedTimestampInterceptor.scala
package org.hibernate.examples.interceptor

import java.util
import org.hibernate.EmptyInterceptor
import org.hibernate.examples.model.UpdatedTimestampEntity

/**
 * 엔티티가 변경된 최신 시각을 관리하는 Interceptor 입니다.
 *
 * @author 배성혁 sunghyouk.bae@gmail.com
 * @since 2013. 12. 24. 오후 4:27
 */
class UpdatedTimestampInterceptor extends EmptyInterceptor {

    override def preFlush(entities: util.Iterator[_]) {
        while (entities.hasNext) {
            val entity = entities.next()
            entity match {
                case ute: UpdatedTimestampEntity => ute.updateUpdatedTimestamp()
                case _ =>
            }
        }
    }
}

File: hibernate-examples/src/main/java/org/hibernate/examples/hibernate/interceptor/UpdatedTimestampInterceptor.java
package org.hibernate.examples.hibernate.interceptor;

import lombok.extern.slf4j.Slf4j;
import org.hibernate.EmptyInterceptor;
import org.hibernate.examples.model.UpdatedTimestampEntity;

import java.util.Iterator;

/**
 * 엔티티의 최신 갱신 시각을 관리하는 Interceptor 입니다.
 *
 * @author 배성혁 sunghyouk.bae@gmail.com
 * @since 2013. 11. 28. 오후 1:50
 */
@Slf4j
public class UpdatedTimestampInterceptor extends EmptyInterceptor {

  /**
   * 엔티티 저장 전에 최신 갱신 시각을 새롭게 갱신합니다.
   */
  @Override
  public void preFlush(Iterator entities) {
    while (entities.hasNext()) {
      Object entity = entities.next();
      if (entity instanceof UpdatedTimestampEntity) {
        ((UpdatedTimestampEntity) entity).updateUpdatedTimestamp();
      }
    }
  }


  private static final long serialVersionUID = -1965299893946137808L;
}
